src/llama-kv-cache.cpp

```diff
@@ -293,26 +348,77 @@ llama_pos llama_kv_cache_unified::seq_pos_max(llama_seq_id seq_id) const {
     return cells.seq_pos_max(seq_id);
 }
 
-void llama_kv_cache_unified::restore() {
-    for (auto & state : recovery.states) {
-        cells.set(state.i, state.cells);
+llama_memory_decode_state_ptr llama_kv_cache_unified::init(
+            const llama_batch & batch,
+            uint32_t n_ubatch,
+            bool embd_pooled,
+            bool logits_all) {
+    GGML_UNUSED(embd_pooled);
+
+    auto sbatch = llama_sbatch(batch, hparams.n_embd, true, logits_all);
+
+    std::vector<llama_ubatch> ubatches;
+    while (sbatch.n_tokens > 0) {
+        ubatches.push_back(sbatch.split_simple(n_ubatch));
```

Comment:
I think I'm reading this right that it's not possible to use `split_equal` with the unified cache after this refactor. If so, I think this will cause problems with the eventual hybrid cache where `split_equal` is required for recurrent child caches (cc @compilade since you [pointed that out to me earlier](https://github.com/ggml-org/llama.cpp/pull/13276#discussion_r2072240777)).